src/create-transformer.ts

```diff
@@ -25,18 +33,23 @@ export function createTransformer<A, B>(
 
     function createView(sourceIdentifier: number, sourceObject: A) {
         let latestValue: B
+        const { onCleanup, debugNameGenerator } = typeof arg2 === 'object' ? arg2 : new Object();
+        const prettifiedName = debugNameGenerator ?
+            debugNameGenerator(sourceObject) :
+            `Transformer-${(<any>transformer).name}-${sourceIdentifier}`;
         const expr = computed(
             () => {
                 return (latestValue = transformer(sourceObject))
             },
             {
-                name: `Transformer-${(<any>transformer).name}-${sourceIdentifier}`
+                name: prettifiedName
             }
         )
         const disposer = onBecomeUnobserved(expr, () => {
             delete views[sourceIdentifier]
             disposer()
             if (onCleanup) onCleanup(latestValue, sourceObject)
+            if (typeof arg2 === 'function') arg2(latestValue, sourceObject)
```

Comment:
I had to read this a few time before I understood it. I would drop line 52, and instead after line 36 add `onCleanup = onCleanup || (typeof arg2 === "function" ? arg2 : undefined)`